src/robot/variables/replacer.py

```diff
@@ -142,12 +142,12 @@ def _get_list_variable_item(self, splitter):
         try:
             index = int(index)
         except ValueError:
-            raise DataError("List variable '%s' used with invalid index '%s'."
+            raise VariableError("List variable '%s' used with invalid index '%s'."
                             % (name, index))
```

Comment:
Second line should be re-indented. Same issue with other VariableErrors in this file too.